scripts/gcloud_adapter.py

```diff
@@ -45,6 +83,57 @@ def update_indexes(index_yaml_path, app_name):
         index_yaml_path, '--project=%s' % app_name])
 
 
+def get_indexes(app_name):
+    """Obtains indexes serving on the server.
+
+    Args:
+        app_name: str. The name of the GCloud project.
+
+    Returns:
+        list. A list of dict of serving indexes.
+    """
+    listed_indexes = subprocess.check_output([
+        GCLOUD_PATH, 'datastore', 'indexes', 'list',
+        '--project=%s' % app_name, '--format=json'])
+    return json.loads(listed_indexes)
+
+
+def check_all_indexes_are_serving(app_name):
+    """Checks that all indexes are serving.
+
+    Args:
+        app_name: str. The name of the GCloud project.
+
+    Returns:
+        bool. A boolean to indicate whenther all indexes are serving or not.
+    """
+    # Serving indexes is a list of dict of indexes. The format of
+    # each dict is as follows:
+    # {
+    #   "ancestor": "NONE",
+    #   "indexId": "CICAgIDAiJ0K",
+    #   "kind": "_AE_Pipeline_Record",
+    #   "projectId": "test-oppia",
+    #   "properties": [
+    #     {
+    #       "direction": "ASCENDING",
+    #       "name": "is_root_pipeline"
+    #     },
+    #     {
+    #       "direction": "DESCENDING",
+    #       "name": "start_time"
+    #     }
+    #   ],
+    #   "state": "READY"
+    # }
+    indexes_serving = get_indexes(app_name)
+    for index in indexes_serving:
```

Comment:
Could simplify to `return all(index['state'] == 'READY' for index in all_indexes)`

Reply:
Done.